src/main/zip/discoverCF.groovy

```diff
@@ -71,13 +100,16 @@ def apiSet = {
         out.flush();
     }
 
-    // parse out url from output
-    int beginIndex = outputStream.indexOf(':') + 2
-    outputStream.delete(0, beginIndex)
-    int endIndex = outputStream.indexOf(' ')
-    outputStream.delete(endIndex, outputStream.length())
+    // parse out api endpoint from output
+    def output = outputStream.toString()
+    output = output.split("\r?\n")[0] // Get first line of output
+    output = output.split(":")
+    apiEndpoint = output[output.length - 1] // Isolate endpoint
```

Comment:
Small... 
but the output.length is not necessary. `-1` is enough.
`apiEndpoint = output[-1] // Isolate endpoint`
Example: http://mrhaki.blogspot.com/2009/03/use-negative-index-in-groovy-lists-to.html